packages/core/src/selectors/base.ts

```diff
@@ -0,0 +1,45 @@
+import { QueryBundle, ToolMetadataOnlyDescription } from "../types";
+
+export interface SingleSelection {
+  index: number;
+  reason: string;
+}
+
+export type SelectorResult = {
+  selections: SingleSelection[];
+};
+
+type QueryType = string | QueryBundle;
+
+function wrapChoice(
+  choice: string | ToolMetadataOnlyDescription,
+): ToolMetadataOnlyDescription {
+  if (typeof choice === "string") {
+    return { description: choice };
+  } else {
+    return choice;
+  }
+}
+
+function wrapQuery(query: QueryType): QueryBundle {
```

Comment:
So we currently don't use the QueryBundle in LITS, and it looks like you're not really using the QueryBundle either since it only has the string, so I would 1. take it out and just use a query string for now, and 2. work with Logan to understand why we might want a QueryBundle in the future (if we actually do).

Reply:
I think that's not a big priority for this PR then we can keep discussing with Logan to understand better and see if it makes sense or if we remove it